game_board/game_board_test.go

```diff
@@ -0,0 +1,108 @@
+package game_board
+
+import(
+	"testing"
+	"math/rand"
+	"fmt"
+	"errors"
+
+	"github.com/stretchr/testify/assert"
+)
+
+
+func TestInitializeBoard(t *testing.T) {
+	fmt.Println("something")
+
+	randomNumber = rand.New(rand.NewSource(1))
+
+	g := GameBoard{BoardDimension: 4 }
+	err := g.InitializeBoard()
+
+	assert.Nil(t, err)
+	assert.Equal(t, 4, len(g.UnmaskedBoard))
+	assert.Equal(t, 4, len(g.UnmaskedBoard[0]))
+
+	expectedBoard := [][]int{[]int{1, 1, 1, 1}, []int{1, 0, 1, 0}, []int{0, 0, 0, 1}, []int{0, 1, 0, 0}}
+	expectedMaskedBoard := [][]string{[]string{ "?", "?", "?", "?"}, []string{"?", "?", "?", "?"}, []string{"?", "?", "?", "?"}, []string{"?", "?", "?", "?"}}
+	assert.Equal(t, expectedBoard, g.UnmaskedBoard)
+	assert.Equal(t, expectedMaskedBoard, g.MaskedBoard)
+}
+
+func TestInitializeGameBoardInvalidBoardDimension(t *testing.T) {
+	g := GameBoard{BoardDimension: -1 }
+
+	err := g.InitializeBoard()
+
+	assert.Error(t, err)
+	assert.Equal(t, errors.New("Board dimension is less than 1"), err)
+}
+
+func TestCheckForHitGopherFound(t *testing.T) {
+	randomNumber = rand.New(rand.NewSource(1))
+
+	g := GameBoard{BoardDimension: 4 }
+	g.InitializeBoard()
+
+	hit, err := g.CheckForHit(0, 0)
+	assert.True(t, true, hit)
+	assert.Nil(t, err)
+}
+
+func TestCheckForHitGopherNotFound(t *testing.T) {
+	randomNumber = rand.New(rand.NewSource(1))
+
+	g := GameBoard{BoardDimension: 4 }
+	g.InitializeBoard()
+
+	hit, err := g.CheckForHit(1, 1)
+
+	assert.False(t, hit)
+	assert.Nil(t, err)
+}
+
+func TestCheckForHitInvalidRowDimension(t *testing.T) {
+	g := GameBoard{BoardDimension: 4 }
+	g.InitializeBoard()
+
+	hit, err := g.CheckForHit(-1, 1)
+
+	assert.False(t, hit)
```

Comment:
tip: Another common go testing pattern is to ignore the returned value when an error is returned. So usually this would look like this:

```go
_, err := g.CheckForHit(-1, 1)

require.EqualError(t, err, "row dimension is not valid")
```